mmv1/products/cloudaicompanion/CodeRepositoryIndex.yaml

```diff
@@ -0,0 +1,110 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: CodeRepositoryIndex
+description: The resource for managing Code Repository Indexes for Gemini Code Assist.
+min_version: 'beta'
+base_url: projects/{{project}}/locations/{{location}}/codeRepositoryIndexes
+self_link: projects/{{project}}/locations/{{location}}/codeRepositoryIndexes/{{code_repository_index_id}}
+create_url: projects/{{project}}/locations/{{location}}/codeRepositoryIndexes?codeRepositoryIndexId={{code_repository_index_id}}
+id_format: projects/{{project}}/locations/{{location}}/codeRepositoryIndexes/{{code_repository_index_id}}
+import_format:
+  - projects/{{project}}/locations/{{location}}/codeRepositoryIndexes/{{code_repository_index_id}}
+examples:
+  - name: "gemini_code_repository_index_basic"
+    min_version: 'beta'
+    primary_resource_id: "example"
+    test_vars_overrides:
+      cri_id: '"cri-example"'
+    exclude_test: true
+autogen_async: true
+async:
+  operation:
+    timeouts:
+      insert_minutes: 90
+      update_minutes: 90
+      delete_minutes: 90
+    path: name
+    base_url: "{{op_id}}"
+    wait_ms: 1000
+  actions:
+    - create
+    - delete
+    - update
+  type: OpAsync
+  result:
+    resource_inside_response: true
+    path: response
+  error:
+    path: error
+    message: message
+  include_project: false
+custom_code:
+  custom_create: templates/terraform/custom_create/code_repository_index.go.tmpl
```

Comment:
As queueing is done on the code repository index level, I suppose we shouldn't ever run into the case of full queue when creating CRI. If there will two creation requests with the same CRI ID, one should fail with Already Exists error.

Reply:
Removed `custom_create` for CRI